test/system/src/test/java/io/pravega/test/system/SegmentReaderAPITest.java

```diff
@@ -0,0 +1,517 @@
+                .disableAutomaticCheckpoints()
+                .startingStreamCuts(startSC)
+                .build();
+        readerGroup.resetReaderGroup(readerGroupConfig);
+
+        //After scale up reading the event
+        reader0 = clientFactory.createReader(readerName,
+                readerGroupName,
+                new UTF8StringSerializer(),
+                ReaderConfig.builder().build());
+        assertEquals(readEvent(reader0), 4);
+
+        ArrayList<SegmentRange> segmentList1 = Lists.newArrayList(batchClient.getSegments(stream, StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+        log.info("Segment List1 :{}", segmentList1);
+
+        Map<Segment, Long> map = segmentList1.stream().collect(Collectors.toMap(SegmentRange::getSegment, value -> value.getEndOffset()));
+        assertTrue(nextStreamCut5 != null);
+        assertTrue(nextStreamCut5.asImpl().getPositions().size() == 2);
+        assertTrue(nextStreamCut5.asImpl().getPositions().containsKey(segment1) &&
+                nextStreamCut5.asImpl().getPositions().containsKey(segment2));
+        assertTrue(map.get(segment1).longValue() <= nextStreamCut5.asImpl().getPositions().get(segment1).longValue());
+        assertTrue(map.get(segment2).longValue() <= nextStreamCut5.asImpl().getPositions().get(segment2).longValue());
+        //Scaling up end
+    }
+
+    /**
+     * This test the getNextStreamCutWithScaleDownTest api with the current streamcut containing one segment0, and written the 5 events.
+     * Created the streamCut0, streamCut1, by calling getNextStreamCut.
+     * Read the events between the streamCut0 and streamCut1 and validated it.
+     * Segment0 has scaled up (segment1 and segment2) with 5 events and validating the numbers of events written
+     * segment1 and segment2 has scaled down (segment3).
+     */
+    @Test(timeout = 60000)
+    public void getNextStreamCutWithScaleDownTest() throws SegmentTruncatedException, ExecutionException, InterruptedException {
+        String streamName = "testStreamSegmentScaleDown";
+        String streamScope = "testScopeSegmentScaleDown";
+        String readerGroupName = "testReaderGroupSegmentScaleDown";
+        String readerName = UUID.randomUUID().toString();
+
+        StreamConfiguration config = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build();
+        ClientConfig clientConfig = ClientConfig.builder()
+                .controllerURI(fetchControllerURI())
+                .build();
+        URI controllerURI = clientConfig.getControllerURI();
+
+        Stream stream = Stream.of(streamScope, streamName);
+
+        @Cleanup
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(streamScope, controller, connectionFactory);
+
+        assertTrue(controller.createScope(streamScope).join());
+        assertTrue(controller.createStream(streamScope, streamName, config).join());
```

Comment:
It's odd this is not going through the public APIs. 
The controller and connectionFactory aren't actually needed in these tests at all, and it can just use the public APIs.